File: Python/find-all-numbers-disappeared-in-an-array.py
class Solution:
    def findDisappearedNumbers(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        for i in range(len(nums)):
            if nums[abs(nums[i]) - 1] > 0:
                nums[abs(nums[i]) - 1] *= -1

        res = list()

        for i in range(len(nums)):
            if nums[i] > 0:
                res.append(i + 1)
            else:
                nums[i] *= -1

        return res

File: Swift/find-all-numbers-disappeared-in-an-array.swift
class Solution {
    func findDisappearedNumbers(_ nums: [Int]) -> [Int] {
        var nums = nums

        for i in 0..<nums.count {
            if nums[abs(nums[i]) - 1] > 0 {
                nums[abs(nums[i]) - 1] *= -1
            }
        }

        var res = [Int]()

        for i in 0..<nums.count {
            if nums[i] > 0 {
                res.append(i + 1)
            } else {
                nums[i] *= -1
            }
        }

        return res
    }
}
